Formula/nbdime.rb

```diff
@@ -0,0 +1,366 @@
+  end
+
+  resource "zipp" do
+    url "https://files.pythonhosted.org/packages/57/dd/585d728479d97d25aeeb9aa470d36a4ad8d0ba5610f84e14770128ce6ff7/zipp-0.6.0.tar.gz"
+    sha256 "3718b1cbcd963c7d4c5511a8240812904164b7f381b647143a89d3b98f9bcd8e"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  def caveats; <<~EOS
+    To enable the git integration run
+      nbdime config-git --enable --global
+  EOS
+  end
+
+  test do
+    (testpath/"old.ipynb").write <<~EOS
+      {
+       "cells": [
+        {
+         "cell_type": "code",
+         "execution_count": null,
+         "metadata": {},
+         "outputs": [],
+         "source": [
+          "print(\\\"Hello World!\\\")"
+         ]
+        }
+       ],
+       "metadata": {
+        "kernelspec": {
+         "display_name": "Python 2",
+         "language": "python",
+         "name": "python2"
+        },
+        "language_info": {
+         "codemirror_mode": {
+          "name": "ipython",
+          "version": 2
+         },
+         "file_extension": ".py",
+         "mimetype": "text/x-python",
+         "name": "python",
+         "nbconvert_exporter": "python",
+         "pygments_lexer": "ipython2",
+         "version": "2.7.10"
+        }
+       },
+       "nbformat": 4,
+       "nbformat_minor": 1
+      }
+    EOS
+    (testpath/"new.ipynb").write <<~EOS
+      {
+       "cells": [
+        {
+         "cell_type": "code",
+         "execution_count": 1,
+         "metadata": {},
+         "outputs": [
+          {
+           "name": "stdout",
+           "output_type": "stream",
+           "text": [
+            "Hello World!\\n"
+           ]
+          }
+         ],
+         "source": [
+          "print(\\\"Hello World!\\\")"
+         ]
+        }
+       ],
+       "metadata": {
+        "kernelspec": {
+         "display_name": "Python 2",
+         "language": "python",
+         "name": "python2"
+        },
+        "language_info": {
+         "codemirror_mode": {
+          "name": "ipython",
+          "version": 2
+         },
+         "file_extension": ".py",
+         "mimetype": "text/x-python",
+         "name": "python",
+         "nbconvert_exporter": "python",
+         "pygments_lexer": "ipython2",
+         "version": "2.7.10"
+        }
+       },
+       "nbformat": 4,
+       "nbformat_minor": 1
+      }
+    EOS
+    diff_output = shell_output("#{bin}/nbdiff --no-color old.ipynb new.ipynb")
+    assert_match "nbdiff old.ipynb new.ipynb", diff_output
+    assert_match /--- old.ipynb  \d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{6}/, diff_output
+    assert_match /\+\+\+ new.ipynb  \d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{6}/, diff_output
+    assert_match "## replaced (type changed from NoneType to int) /cells/0/execution_count:", diff_output
+    assert_match "-  None", diff_output
+    assert_match "+  1", diff_output
+    assert_match "## inserted before /cells/0/outputs/0:", diff_output
```

Comment:
Awesome tests! One minor concern: Is the diff output you're matching on considered completely stable? i.e., it won't change in future releases of nbdime? In general homebrew tests need to find a decent compromise between being a good canary to indicate breakage, and being future proof & easy to maintain.

Reply:
Sorry, I don't really know if that's the case. I would think it should be stable - but I am not a maintainer off the software. I would think though, that this test is probably a little bit overkill - if required we can reduce it to test for the line that check the `----` and `+++` lines, as these conform to diff output formats and are very likely to be stable.

Reply:
Yes, I think it would be good to check output for stable stuff, both from the perspective of changes to `nbdime` and possible future changes to notebook format. Ideally, you should diff two minimal differing notebooks and look for something that won't change in the output. Also, I don't know if nbdime uses return status to indicate differences  or lack thereof, but it might be an equally useful test to check that there are __no__ differences when you compare two identical files.